### PR TITLE
Change constants to immutable format.

### PR DESCRIPTION
**Pull request by iteyelmp**
immutable can be initialized in the constructor, and the value will be used as part of the bytecode.

old constants

```
 17, // maxKvSizeBits, 131072
 39, // shardSizeBits ~ 512G
 2, // randomChecks
 4718592000, // minimumDiff 5 * 3 * 3600 * 1024 * 1024 / 12 = 4718592000 for 5 replicas that can have 1M IOs in one epoch
 7200, // cutoff = 2/3 * target internal (3 hours), 3 * 3600 * 2/3
 32, // diffAdjDivisor
 100, // treasuryShare, means 1%

 startTime, // startTime
 1500000000000000, // storageCost - 1,500,000Gwei forever per blob - https://ethresear.ch/t/ethstorage-scaling-ethereum-storage-via-l2-and-da/14223/6#incentivization-for-storing-m-physical-replicas-1
 340282366367469178095360967382638002176n, // dcfFactor, it mean 0.95 for yearly discount
 1048576, // nonceLimit 1024 * 1024 = 1M samples and finish sampling in 1.3s with IO rate 6144 MB/s: 4k * 2(random checks) / 6144 = 1.3s
 treasuryAddress, // treasury
 3145728000000000000000n, // prepaidAmount - 50% * 2^39 / 131072 * 1500000Gwei, it also means 3145 ETH for half
```

put blob gas use:
https://dashboard.tenderly.co/tx/sepolia/0x0364e1e03435154398f25c47db94676e8cf00c14ad1980459c60ccdb2271d182/gas-usage

replace blob gas use:
https://dashboard.tenderly.co/tx/sepolia/0x59cf5e73c2e06cc8631382f463d43f53524155eba1419ed96e3e36f63f6809ad/gas-usage

### CODE REVIEW

File: contracts/StorageContract.sol [hunk 1]
```diff
@@ -9,10 +9,61 @@ import "./RandaoLib.sol";
...
+    }
+
+    uint256 public constant sampleSizeBits = 5; // 32 bytes per sample
+
+    uint256 public immutable maxKvSizeBits;
+    uint256 public immutable shardSizeBits;
+    uint256 public immutable shardEntryBits;
+    uint256 public immutable sampleLenBits;
+    uint256 public immutable randomChecks;
+    uint256 public immutable minimumDiff;
```
Comment (qzhodl) [inline contracts/StorageContract.sol:29]: nonceLimit, minimumDiff and prepaidAmount are not related to the gas cost of putBlob, and I think we can change them to public variables and add only owner setter func.

File: contracts/DecentralizedKV.sol [hunk 2]
```diff
@@ -4,16 +4,21 @@ pragma solidity ^0.8.0;
 import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
 import "./MerkleLib.sol";
 import "./BinaryRelated.sol";
-import "./EthStorageConstants.sol";
```
Comment (qzhodl) [inline contracts/DecentralizedKV.sol:7]: We should also have a test gas cost result for putBlob func: case 1: putBlob for a new key; case 2: update an existing key

Comment (qzhodl) [reply contracts/DecentralizedKV.sol:7]: need a test gas cost for updating an existing key value

File: scripts/deploy.js [hunk 3]
```diff
@@ -1,27 +1,53 @@
...
+  const startTime = Math.floor(new Date().getTime() / 1000);
+
   const [deployer] = await hre.ethers.getSigners();
   ownerAddress = deployer.address;
   treasuryAddress = deployer.address;
 
   const StorageContract = await hre.ethers.getContractFactory("TestEthStorageContractKZG");
   // refer to https://docs.google.com/spreadsheets/d/11DHhSang1UZxIFAKYw6_Qxxb-V40Wh1lsYjY2dbIP5k/edit#gid=0
-  const implContract = await StorageContract.deploy({ gasPrice: gasPrice });
+  const implContract = await StorageContract.deploy([
```
Comment (syntrust) [inline scripts/deploy.js:31]: Just a suggestion: reduce the duplicated appearances of the params
```
[
        17, // maxKvSizeBits, 131072
        39, // shardSizeBits ~ 512G
        2, // randomChecks
        7200, // cutoff = 2/3 * target internal (3 hours), 3 * 3600 * 2/3
        32, // diffAdjDivisor
        100, // treasuryShare, means 1%
      ]
```

**Review by syntrust — APPROVED** [review]

**Review by qzhodl — APPROVED** [review]

**Pull request merged**